### ISSUE TITLE
When is StartCopyFromBlob applicable?

### ISSUE THREAD

**Issue by xraywu** [Azure/azure-storage-net#150]
Hi I am trying to use **StartCopyFromBlob** to download files directly to my storage account. In most of the cases it works fine however when I try to download a video from youtube, it throws me this error - 

```C#
// The URL is the actual URL pointing to a youtube video that is open to public
var url = @"http://r2---sn-a5m7zu7e.googlevideo.com/videoplayback?source=youtube&dur=288.345&ip=202.126.172.38&ipbits=0&pl=27&key=cms1&signature=52C561D1DDC14A0716C909522E68577A52336974.49F877BC54AD80CBB671F09432D4846CACAA6487&id=o-AFRztEln1pr06gyTNLXru0klWglUWHDuYhyEi9CamCaB&sparams=dur,expire,id,ip,ipbits,itag,lmt,mime,mm,mn,ms,mv,nh,pcm2cms,pl,ratebypass,source,upn&sver=3&upn=1kpiikLgCAY&mime=video%2Fmp4&itag=22&fexp=901816%2C9408142%2C9408213%2C9408420%2C9408710%2C9408737%2C9410705%2C9415660%2C9415956%2C9416126%2C9416222%2C9416729%2C9416768%2C9416884&lmt=1399394881143966&expire=1436975700&ratebypass=yes&req_id=7a82f6ae33d7a3ee&redirect_counter=2&cms_redirect=yes&mm=26&mn=sn-a5m7zu7e&ms=tsu&mt=1436954183&mv=m"

var blockBlob = BlobContainer.GetBlockBlobReference("blobName");
blockBlob.StartCopyFromBlob(new Uri(url), null, null, null);
```

This throws me the following error - 

    Unhandled Exception: System.AggregateException: One or more errors occurred. ---> 
    Microsoft.WindowsAzure.Storage.StorageException: The remote server returned an error: (400) Bad Request. ---> 
    System.Net.WebException: The remote server returned an error: (400) Bad Request.
       at System.Net.HttpWebRequest.GetResponse()
       at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ExecuteSync[T](RESTCommand`1 cmd, IRetryPolicy policy, OperationContext operationContext)
       --- End of inner exception stack trace ---
       at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ExecuteSync[T](RESTCommand`1 cmd, IRetryPolicy policy, OperationContext operationContext)
       at Microsoft.WindowsAzure.Storage.Blob.CloudBlockBlob.StartCopyFromBlob(Uri source, AccessCondition sourceAccessCondition, AccessCondition destAccessCondition, BlobRequestOptions options, OperationContext operationContext)

It seems certain requests can not get through...

**Comment by pemari-msft** [comment]
Hi @xraywu, I'll take a look at this today, but firstly, are you sure that URL is correct? I tried clicking on the link and got a 403 Forbidden error.

**Comment by xraywu** [comment]
@pemari-msft sorry the link got expired - however at the time when I tried to download it from Azure Storage it was still valid. You can always use http://en.savefrom.net/ and enter a link from youtube (e.g. https://www.youtube.com/watch?v=5ndj6xn2P0c) to get a valid url at the moment.

**Comment by pemari-msft** [comment]
Hi @xraywu. Thanks for reporting this. We actually identified a bug in our server-side logic where some characters in a host name in a certain case cause the request to fail during a copy. We've logged the bug and will look to fix it in a future Azure Storage Service Version release. Sorry for the inconvenience. 

As a temporary workaround, resolving the host name to an ip address and using that ip address as your copy source should unblock you.
